test/intl402/NumberFormat/prototype/format/value-decimal-string.js

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-number-format-functions
+description: >
+  Intl.NumberFormat.prototype.format converts its argument (called value) to a
+  number using ToIntlMathematicalValue.
+features: [Intl.NumberFormat-v3]
+locale: [en-US]
+---*/
+
+var nf = new Intl.NumberFormat('en-US');
+
+// The value 100,000 should only be interpreted as infinity if the input is the
+// string "Infinity".
+assert.sameValue(nf.format('100000'), '100,000');
+// The value -100,000 should only be interpreted as negative infinity if the
+// input is the string "-Infinity".
+assert.sameValue(nf.format('-100000'), '-100,000');
```

Comment:
I don't understand what invariant these tests are verifying.  Is the number 100,000 significant in the spec somewhere?

Reply:
This is my attempt to verify the following conditions in ToIntlMathematicalValue:

> 8. If mv is 10<sup>10000</sup> and str contains Infinity, return +∞.
> 9. If mv is -10<sup>10000</sup> and str contains Infinity, return -∞.

The intent of these assertions is to ensure that implementations reference not only the mathematical value of the input but also the original string representation. I believe these assertions would fail if an engine incorrectly implemented somethng like this:

> 8. If mv is 10<sup>10000</sup>, return +∞.
> 9. If mv is -10<sup>10000</sup>, return -∞.

Does that seem right to you?